packages/dataviews/src/components/dataviews-picker/index.tsx

```diff
@@ -0,0 +1,267 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { __experimentalHStack as HStack } from '@wordpress/components';
+import {
+	useContext,
+	useEffect,
+	useMemo,
+	useRef,
+	useState,
+} from '@wordpress/element';
+import { useResizeObserver, throttle } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import DataViewsContext from '../dataviews-context';
+import {
+	default as DataViewsFilters,
+	useFilters,
+	FiltersToggle,
+} from '../dataviews-filters';
+import DataViewsLayout from '../dataviews-layout';
+import { DataViewsPickerFooter } from './footer';
+import DataViewsSearch from '../dataviews-search';
+import { DataViewsPagination } from '../dataviews-pagination';
+import DataViewsViewConfig, {
+	DataviewsViewConfigDropdown,
+	ViewTypeMenu,
+} from '../dataviews-view-config';
+import { normalizeFields } from '../../normalize-fields';
+import type { Action, Field, View, SupportedLayouts } from '../../types';
+import type { SelectionOrUpdater } from '../../private-types';
+type ItemWithId = { id: string };
+
+const isItemClickable = () => false;
+
+type DataViewsPickerProps< Item > = {
+	view: View;
+	onChangeView: ( view: View ) => void;
+	fields: Field< Item >[];
+	actions?: Action< Item >[];
```

Comment:
This is mostly for my own education: have you thought of making actions a list of `ActionButton<Item>[]`? They're already restricted, and it's fine. I appreciate we may provide a built-in replacement for DataViews and the component will ignore the modal actions.

I suppose the advantage of the current approach is that the consumer doesn't have to deal with removing the modal actions. For example, they can get the actions for an entity in the same ways DataViews-screen do. Or situations where DataViews and DataViewsPicker are used in the same page.

Changing the type to an ActionButton list would provide the consumer type information. Useful, but less beneficial than just reusing the actions, I guess?

Reply:
I like the suggestion. I'll change it to ActionButton<Item>[]. Easy to change again if we need to,